### PR TITLE
[CL-278][CL-391] misc bit-item style fixes

### PR DESCRIPTION
**Pull request by willmartian**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/CL-278

https://bitwarden.atlassian.net/browse/CL-391

## 📔 Objective

`bit-item` misc style fixes:

- use correct colors
- use small icon button in stories
- use pointer when hovering links/anchors

## 📸 Screenshots

See Storybook

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: libs/components/src/item/item.component.html [hunk 1]
```diff
@@ -1,9 +1,8 @@
-<!-- TODO: Colors will be finalized in the extension refresh feature branch -->
 <div
-  class="tw-box-border tw-overflow-auto tw-flex tw-bg-background [&:has(.item-main-content_button:hover,.item-main-content_a:hover)]:tw-bg-primary-300/20 tw-text-main tw-border-solid tw-border-b tw-border-0 tw-rounded-lg tw-mb-1.5"
+  class="tw-box-border tw-overflow-auto tw-flex tw-bg-background [&:has(.item-main-content_button:hover,.item-main-content_a:hover)]:tw-cursor-pointer [&:has(.item-main-content_button:hover,.item-main-content_a:hover)]:tw-bg-primary-100 tw-text-main tw-border-solid tw-border-b tw-border-0 tw-rounded-lg tw-mb-1.5"
```
Comment (vleague2) [inline libs/components/src/item/item.component.html:2]: Woah is this like an `or` selector for the `has`?? 

Also, just to double-check, the `cursor-pointer` works in the extension too?

**Review by vleague2 — COMMENTED** [review]
The ticket also mentions something about the gap between buttons, do we need to do anything there?

**Comment by willmartian** [comment]
> The ticket also mentions something about the gap between buttons, do we need to do anything there?

Nope! That must have been already fixed.

File: libs/components/src/item/item.component.html (hunk 1 above)
Comment (willmartian) [reply libs/components/src/item/item.component.html:2]: > Woah is this like an or selector for the has??

Yeah, `,` is the OR combinator in CSS, and `:has` allows for any selector combinators within it. 

> the cursor-pointer works in the extension too?

Yep!

**Review by vleague2 — APPROVED** [review]

**Pull request merged**